### PR TITLE
Rework function parser helpers into symbol parser helpers that parse functions, variables, and classes

### PR DESCRIPTION
**Pull request by wldcordeiro**
Associated Issue: #2249 

### Summary of Changes

* Rework `getFunctions` into `getSymbols` function that traverses an AST and extracts variable, function and class declaration names and locations

### Test Plan

- [x] Run the unit tests and see that they're passing
- [x] Test that function search is still working as expected

### CODE REVIEW

File: src/utils/tests/parser.js [hunk 1]
```diff
@@ -57,8 +58,15 @@ const classTest = formatCode(`
   }
 `);
 
+const varTest = formatCode(`
+  var foo = 1;
+  let bar = 2;
+  const baz = 3;
+  const a = 4, b = 5;
+`);
```
Comment (jasonLaster) [inline src/utils/tests/parser.js]: perhaps we should find params and properties as well:

`function(bar) {}`
`x = {a: 2}`

Comment (wldcordeiro) [inline src/utils/tests/parser.js]: Agreed! Added parameter parsing. Now just properties to go 😄

**Comment by wldcordeiro** [comment]
@jasonLaster if you have any ideas for additional tests it'd be much appreciated. 😄 Also since our tests mostly check for names we don't handle multiple symbols of the same name well in the tests. The code returns objects with names and locations though so that shouldn't affect our function/variable search utils themselves.

**Comment by wldcordeiro** [comment]
Adding class parsing was easy so I did that too. 😃

File: src/utils/parser.js [hunk 2]
```diff
@@ -91,40 +111,90 @@ function isFunction(path) {
...
     value: dec.name,
     location: dec.location
   }));
+}
+
+function getSymbolDeclarations(
+  sourceText: SourceText): FormattedSymbolDeclarations {
+  if (symbolDeclarations.has(sourceText.id)) {
+    const symbols = symbolDeclarations.get(sourceText.id);
+    if (symbols) {
```
Comment (wldcordeiro) [inline src/utils/parser.js]: Flow complained without this guard here. 😆

File: src/utils/parser.js [hunk 3]
```diff
@@ -91,40 +111,90 @@ function isFunction(path) {
...
+  return declarations.map(dec => ({
     id: `${dec.name}:${dec.location.start.line}`,
     title: dec.name,
     subtitle: `:${dec.location.start.line}`,
     value: dec.name,
     location: dec.location
   }));
+}
+
+function getSymbolDeclarations(
```
Comment (wldcordeiro) [inline src/utils/parser.js]: Maybe kill this function in place of doing this work within `getSymbols`? I tried keeping it as similar to the previous API.

**Pull request merged**